packages/react-native/Libraries/Core/setUpTimers.js

```diff
@@ -50,6 +52,31 @@ if (global.RN$Bridgeless !== true) {
   defineLazyTimer('cancelIdleCallback');
 }
 
+if (
+  NativeReactNativeFeatureFlags != null &&
+  // eslint-disable-next-line react-hooks/rules-of-hooks -- false positive due to `use` prefix
+  ReactNativeFeatureFlags.useModernRuntimeScheduler()
+) {
+  polyfillGlobal(
+    'requestIdleCallback',
+    () => NativeIdleCallbacks.requestIdleCallback,
+  );
+
+  polyfillGlobal(
+    'cancelIdleCallback',
+    () => NativeIdleCallbacks.cancelIdleCallback,
+  );
+} else {
+  polyfillGlobal(
+    'requestIdleCallback',
+    () => require('./Timers/JSTimers').requestIdleCallback,
+  );
+  polyfillGlobal(
+    'cancelIdleCallback',
+    () => require('./Timers/JSTimers').cancelIdleCallback,
+  );
+}
+
```

Comment:
This should be part of an `else` block in the previous if instead (basically, if we're in bridgeless).